### DISCUSSION TITLE
Can the application store one of the key shares on the application's server?

### DISCUSSION THREAD

**Discussion by nakulmk** [Web3Auth/Web3Auth#345] [Help]
Can I customize the key splitting?

**Comment by himanshuchawla009** [comment]
Yes, if you are using a custom verifier, then web3auth returns a dappShare which the application can store and manage.  Web3Auth's `getUserInfo` function returns this dapp share.

Refer here to know more  about getUserInfo function : https://web3auth.io/docs/sdk/web/web3auth/usage#get-users-information

Also, you can pass this share while login in using web3auth's open login adapter login params. 
Refer here to know more about login params:- https://web3auth.io/docs/sdk/web/core/usage#loginparams

**Reply by sandys** [reply]
@himanshuchawla009 we also need this. can u explain a bit more ? i thought to achieve this - we needed to use the self-hosted SDK (tkey) directly and control the key splitting. But we are fairly lost on how to go about doing this.
Unsure if this part comes in the custom authentication part or the custom verifier part or something in tkey.

one of the confusion is that a custom verifier seems to be something deployed on the blockchain. however both for previous question (and us), we just have a backend api where we want to keep one of the shares

**Reply by himanshuchawla009** [reply]
we return a share for custom verifiers back to application (dappShare) which you can store as a backup, plz go through this answer for more information about dappShare  : https://github.com/orgs/Web3Auth/discussions/427#discussioncomment-3442052
